store/src/server/api.rs

```diff
@@ -1,14 +1,18 @@
-use std::net::ToSocketAddrs;
-use tracing::info;
-use anyhow::Result;
-
 use crate::config::StoreConfig;
-use tonic::{transport::Server, Response, Status, Streaming};
-
+use crate::db::interface::Db;
+use anyhow::Result;
+use deadpool_sqlite::Pool;
+use miden_crypto::merkle::TieredSmt;
+use miden_crypto::{Felt, FieldElement};
 use miden_node_proto::store;
+use std::net::ToSocketAddrs;
+use tonic::{transport::Server, Response, Status, Streaming};
+use tracing::info;
 
-#[derive(Default)]
-pub struct DBApi {}
+pub struct DBApi {
+    pool: Pool,
+    nullifier_tree: TieredSmt,
+}
```

Comment:
minor nit: should we rename `pool` into `db`? Whether it is a pool or a single connection seems like an implementation detail.

Also, would a better name be `StoreApi` rather than `DbApi`?

Reply:
> minor nit: should we rename pool into db? Whether it is a pool or a single connection seems like an implementation detail.

I can rename if you want, but this is a pool and not a connection, and it does not behave like a connection.

> Also, would a better name be StoreApi rather than DbApi?

Renamed